### ISSUE TITLE
Diagnostics only show R Markdown chunk errors (visual mode)

### ISSUE THREAD

**Issue by ronblum** [rstudio/rstudio#10985]
### System details

    RStudio Edition : Desktop and Server
    RStudio Version : 2022.06.0+296
    OS Version      : MacOS 12.3 and Ubuntu 20.04
    R Version       : 4.1.3 & 4.2.0

### Steps to reproduce the problem

1. Turn on diagnostics in <kbd>Tools -> Global Options -> Code -> Diagnostics</kbd>
2. Open a new R Markdown file; the default works.
3. Switch to visual mode.
4. In a chunk, add some bogus code—an undefined object, syntax error, etc. E.g.
````
```{r cars}
summary(cars)
y
print(
```
````
5. Save the file.

### Describe the problem in detail

Diagnostic information showing the syntax error for `print(` is displayed. The informational message about the undefined object does not, though.

<img width="194" alt="diag_missing" src="https://user-images.githubusercontent.com/31009952/163439018-87af0ae0-d80a-489e-86de-8445a73b922e.png">

### Describe the behavior you expected

The lines with the problems should be identified in place, as it does with Prairie Trillium 2022.02.2+481.

<img width="177" alt="diag_there" src="https://user-images.githubusercontent.com/31009952/163439007-03ec769b-f12b-4f55-8e6a-47aac5c7fdca.png">

- [ ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
Actually, this might be a variation of #10984, since the errors do show up in the wrong place if I do the following:
````
```{r cars}
summary(cars)
x <- 1
x
x
x
x
x
x
x
x
x
y
print(
```
````

<img width="208" alt="maybe_the_same" src="https://user-images.githubusercontent.com/31009952/163462148-b7c3b57f-4329-4e5e-bd8a-6a8d972a603f.png">

**Comment by jgutman** [comment]
@ronblum agree with your assessment, as this now seems to be fixed after the fix for #10984

<img width="916" alt="image" src="https://user-images.githubusercontent.com/7817881/167944701-77c3a171-9910-49ab-9a13-e82e061013e9.png">

(not yet available in dailies but should be available in SW build 389 or so)

**Comment by romainfrancois** [comment]
Confirmed with RStudio 2022.06.0-daily+391 "Spotted Wakerobin": 

<img width="358" alt="image" src="https://user-images.githubusercontent.com/2625526/168029352-97a79822-8668-4bf8-ab16-04e9a5d53e6e.png">

<img width="308" alt="image" src="https://user-images.githubusercontent.com/2625526/168029480-6bb3685d-8876-4f3a-8285-a2a5481d203a.png">
